src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilter.java

```diff
@@ -157,14 +171,114 @@ private String getFileName(List<String> singlePatch) {
         return fileName;
     }
 
+    private List<Integer> getRemovePositionList(Patch<String> patch, int index) {
+        final Chunk originChunk = patch.getDeltas().get(index).getSource();
+        return originChunk.getChangePosition();
+    }
+
+    private List<Integer> getAddPositionList(Patch<String> patch, int index) {
+        final Chunk targetChunk = patch.getDeltas().get(index).getTarget();
+        return targetChunk.getChangePosition();
+    }
+
     private List<List<Integer>> getLineRange(Patch<String> patch) {
         final List<List<Integer>> lineRangeList = new ArrayList<>();
+        int totalAddNum = 0;
+        int totalRemoveNum = 0;
         for (int i = 0; i < patch.getDeltas().size(); i++) {
-            final Chunk targetChunk = patch.getDeltas().get(i).getTarget();
-            final List<Integer> lineRange = new ArrayList<>();
-            lineRange.add(targetChunk.getPosition());
-            lineRange.add(targetChunk.getPosition() + targetChunk.getLines().size());
-            lineRangeList.add(lineRange);
+            final List<Integer> removePositionList = getRemovePositionList(patch, i);
+            final List<Integer> addPositionList = getAddPositionList(patch, i);
+            if (add) {
+                final List<Integer> newPositionList = new ArrayList<>();
+                int addIndex = 0;
+                int removeIndex = 0;
+                while (addIndex < addPositionList.size()) {
```

Comment:
It is hard to understand what is going on in this loop. It looks like the code is duplicated a lot. Please split the logic into separate meaningful methods.

Reply:
done